### ISSUE TITLE
Warn players when disconnecting from loosing game

### ISSUE THREAD

**Issue by TheJJ** [SFTtech/openage#494]
It's very annoying that players leave the game, especially in early phases, because they think they have lost.

We should add a warning that says something like:
```
Hey you, have you tried fleeing to one of your allies?
Try moving to their base and support their fight!
It's bad habit leaving the other players hanging, so please reconsider.
```

And probably adapt the message by analyzing the game state.
Other ideas? :)

**Comment by eaterjolly** [comment]
It would probably be a better idea to develop upon the ranking system, and penalize severely early game leaves (and perhaps add a mechanism of offering a resignation to mitigate the damage from quitting). I always felt that age of empires aspired to be like chess in it's tournament status, and lots of people have played the game competitively. Most tournament formats try to discourage early game resignations anyway, so it would make sense to implement a penalty in that format. 

I would say to try and implement a chess-based ranking system too, as homage to that original vision.

**Comment by TheJJ** [comment]
Indeed, very good idea actively penalizing early leaves. But mind, the ranking stuff must be n vs n capable, not just 1on1 like chess requires. The ranking mechanism etc is part of the [master server](https://github.com/SFTtech/openage-masterserver).

**Comment by elnabo** [comment]
But wouldn't early rush/aggression trigger valid early leaves ?

**Comment by coffenbacher** [comment]
I don't play / follow the games but I think LoL and Dota have been experimenting with how to penalize abandonment in n vs. n games for years, so when it comes time to decide on a method I think that would be a good place to look for potential solutions.
